AxonDeepSeg/data_management/input_data.py

```diff
@@ -1,437 +1,131 @@
-        :param mask: Mask of the image (ndarray) to apply data augmentation to.
-        :param augmented_data: Dict, contains the parameters of the data augmentation to apply.
-        :param data_aug_verbose: Int. If >=1, displays information about the data augmentation process.
-        :return: Image and Mask that have been transformed.
-        """
+        return (image, mask)
 
+    def on_epoch_end(self):
+        pass
 
-        # Online data augmentation
-        if augmented_data['type'].lower() == 'all':
-            augmented_data.pop('type')
-            augmented_element = all_transformations(element,
-                                                transformations=augmented_data,
-                                                thresh_indices=self.thresh_indices,
-                                                verbose=data_aug_verbose)
+    def __len__(self):
+        return int(np.ceil(len(self.ids) / float(self.batch_size)))
 
-        elif augmented_data['type'].lower() == 'random':
-            augmented_data.pop('type')
-            augmented_element = random_transformation(element,
-                                                  transformations=augmented_data,
-                                                  thresh_indices=self.thresh_indices,
-                                                  verbose=data_aug_verbose)
-
-        else:
-            augmented_element = element
-
-        return augmented_element
-
-    def generate_boundary_weights(self, real_mask, weights_intermediate, sigma):
-        """
-        Generates the boundary weights from the mask.
-        :param real_mask: the discretized mask.
-        :return: The 3D ndarray of the boundary weights (H,W,C) with C the number of classes.
-        """
-
-        # Create a weight map for each class (background is the first class, equal to 1
-        n_classes = len(self.thresh_indices)
-
-        # Classical method to compute weights
-        for indice, class_ in enumerate(self.thresh_indices[1:]):
-
-            mask_class = real_mask[:, :, indice]
-            mask_class_8bit = np.asarray(255 * mask_class, dtype='uint8')
-            weight = ndimage.distance_transform_edt(mask_class_8bit)
-            weight[weight == 0] = np.max(weight)
-
-            if class_ == self.thresh_indices[1]:
-                w0 = 0.5
-            else:
-                w0 = 1
-
-            weight = 1 + w0 * np.exp(-(weight.astype(np.float64) / sigma) ** 2 / 2)
-            weights_intermediate[:, indice] = weight.reshape(-1, 1)[:, 0]
-
-        # Generating the mask with the real labels as well as the matrix of the weights
-        return np.reshape(weights_intermediate, [real_mask.shape[0], real_mask.shape[1], n_classes])
 
+def labellize_mask_2d(patch, thresh_indices=[0, 0.2, 0.8]):
+    '''
+    Process a patch with 8 bit pixels ([0-255]) so that the pixels between two threshold values are set to the closest threshold, effectively
```

Comment:
Since you require that this patch is between [0-255], it would be a good practice to check this and throw an [exception](https://docs.python.org/3/tutorial/errors.html) so that it can be handled if this isn't the case.